safe-discriminant-derive/src/lib.rs

```diff
@@ -80,13 +80,30 @@ fn validate_all_variants(variants: impl Iterator<Item = Variant>) -> Result<()>
         .unwrap_or(Ok(()))
 }
 
+/// Returns true if there is any #[x] where x is not `repr`
+/// returns false otherwise.
+fn contains_attribute_macros(attrs: &[Attribute]) -> bool {
```

Comment:
I think if we want to get super technical, what we really want to say is that there are no attributes past the derive macro that is implementing the `Discriminant` trait.

I actually think that's what this code accomplishes, since I think that the other attribute macros will be consumed in-order, so I don't think you can see the attributes after they've taken effect. But the chat logs that you pointed to [seem to suggest otherwise](https://rust-lang.zulipchat.com/#narrow/stream/122651-general/topic/feedback.20request.3A.20safe.20discriminant.20extraction.20from.20enums/near/463033438) and that you can see other attribute macros after they've been expanded. Am I reading this right?

Anyways, consider this an optional request, but it might be nice to try to create an attribute macro that isn't malicious and does nothing and see if it gets compiled out so that we don't see it as long as it's positioned before the derive attribute. And if so, we can test a non-malicious attribute macro to see that it produces what we expect (an error if placed after the derive attribute, success if placed before).

I'm not totally sure what would happen, but maybe worth playing with.